src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,119 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+
+    /**
+     * Constructor for TaskList
+     *
+     * @param tasks List of tasks to be stored in TaskList
+     */
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public TaskList() {
+        this(new ArrayList<>());
+    }
+
+    /**
+     * Finds tasks with keyword in description and prints to output
+     *
+     * @param keyword String keyword to search for in Task
+     */
+    public void findTasks(String keyword) {
+        TaskList temp = new TaskList();
+        for (Task task : tasks) {
+            if (task.containsKeyword(keyword)) {
+                temp.tasks.add(task);
+            }
+        }
+        System.out.println("Here are the matching tasks in your list:");
+        temp.showTasksPlain();
+    }
+
+    /**
+     * Shows the tasks in TaskList without additional comments
+     */
+    private void showTasksPlain() {
```

Comment:
Maybe you can consider renaming this function for better clarity based on what your code logic for this function is.
```suggestion
    private void showNumberedTasks() {
```